presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchPageSourceProvider.java

```diff
@@ -53,10 +58,33 @@ public ConnectorPageSource createPageSource(
         requireNonNull(split, "split is null");
         requireNonNull(table, "table is null");
 
-        return new ElasticsearchPageSource(
+        ElasticsearchTableHandle elasticsearchTable = (ElasticsearchTableHandle) table;
+        ElasticsearchSplit elasticsearchSplit = (ElasticsearchSplit) split;
+
+        if (columns.isEmpty()) {
+            long count = client.count(
+                    elasticsearchSplit.getIndex(),
+                    elasticsearchSplit.getShard(),
+                    buildSearchQuery(session, elasticsearchTable.getConstraint().transform(ElasticsearchColumnHandle.class::cast), elasticsearchTable.getQuery()));
+
+            if (elasticsearchTable.getLimit().isPresent()) {
+                count = Math.min(elasticsearchTable.getLimit().getAsLong(), count);
+            }
+
+            ImmutableList.Builder<Page> pages = ImmutableList.builder();
+            while (count > 0) {
+                int batch = Ints.saturatedCast(count);
+                pages.add(new Page(batch));
```

Comment:
@sopel39 is it safe to inject 2 billion row pages into the system?

in case of `select count(*) from ..`, they are going to be partial aggregated quickly
what about `select 1 from ..` query?

(Martin, if you make pages, smaller, make it also lazy, as `pages` list  may grow big. Currently, it's well consttrained by my imagination)

Reply:
These pages are empty, actually. The only thing they carry is a count.

Reply:
Yes, I know. But if you project a column (random or constant), they become non-empty and _i think_ ScanFilterProject won't be able to split this into multiple small ones. Will it?